From 9fca9ad69f9b0bf68f955c1ff0ebbe6847497482 Mon Sep 17 00:00:00 2001
From: Gokul Sriram Palanisamy <quic_gokulsri@quicinc.com>
Date: Thu, 24 Oct 2024 16:09:47 +0530
Subject: [PATCH] bus: mhi: host: forbid PM runtime during MISSION MODE

if PM is enabled during MISSION mode, mhi enters M3 on idle.
While in M3, any QMI messages from SDX like DSD is not received.
Because of this, SDX crash/reboot goes unnotified and SSR gets impacted.

Change-Id: I90fa3d9922c5369cdeca332414ad0d005ced860a
Signed-off-by: Gokul Sriram Palanisamy <quic_gokulsri@quicinc.com>
---
 drivers/bus/mhi/host/pci_generic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bus/mhi/host/pci_generic.c b/drivers/bus/mhi/host/pci_generic.c
index 51e45f6c50b5..a027de594043 100644
--- a/drivers/bus/mhi/host/pci_generic.c
+++ b/drivers/bus/mhi/host/pci_generic.c
@@ -735,7 +735,7 @@ static void mhi_pci_status_cb(struct mhi_controller *mhi_cntrl,
 		pm_runtime_forbid(&pdev->dev);
 		break;
 	case MHI_CB_EE_MISSION_MODE:
-		pm_runtime_allow(&pdev->dev);
+		pm_runtime_forbid(&pdev->dev);
 		break;
 	default:
 		break;
-- 
2.34.1

